### PR TITLE
Supp-session-id<-> SessionId should be Unique

### PR DESCRIPTION
**Pull request by tanxeel90**
SessionId should have a unique supp-session-Id. Removed bimap as it was not maintaining uniqueness on AddOrUpdate. Reversed the key of cluster map to SessionId.

### CODE REVIEW

**Comment by tanxeel90** [comment]
I think the Supp-session-id as sessionData is a bit pointless and repetitive now. Probably it should be removed from the sessionData. The cluster map SessionId->SuppSession should be enough to persist the relationship. Comments?

**Comment by StanJav** [comment]
> I think the Supp-session-id as sessionData is a bit pointless and repetitive now. 

Do we use the auto-sync on 'sassionData' change to sync supp-session-id(s)? I do not think so. In that case I agree it is actually pointless.

File: Echo.Process/Session/SessionSync.cs [hunk 1]
```diff
@@ -108,7 +108,7 @@ public int Incoming(SessionAction incoming)
         /// <param name="sessionId"></param>
         /// <returns></returns>
         internal Option<SessionId> GetSessionId(SupplementarySessionId sessionId) =>
-            suppSessions.Find(sessionId);
+            suppSessions.Where(sid => sid == sessionId).Map(t => t.Key).HeadOrNone();
```
Comment (louthy) [inline Echo.Process/Session/SessionSync.cs]: This is far too slow if there are 100,000s of sessions.  We need a better solution.

Comment (tanxeel90) [reply Echo.Process/Session/SessionSync.cs]: Maybe two maps then - SessionId->SuppSess and SuppSess->SessionId. Probably the same goes for cluster implementation: SessionManager.cs line 182 in this PR

**Comment by tanxeel90** [comment]
> > I think the Supp-session-id as sessionData is a bit pointless and repetitive now.
> 
> Do we use the auto-sync on 'sassionData' change to sync supp-session-id(s)? I do not think so. In that case I agree it is actually pointless.

We actually do sync. I was suggesting that we don't pollute ```sessionData``` considering that ```sessionData``` is for the users and ```supp-session-id``` is more echo's own feature so have an independent implementation of Supp-session-id with a sync call only for supp-sessions

File: Echo.Process/Session/SessionSync.cs (hunk 1 above)
Comment (tanxeel90) [reply Echo.Process/Session/SessionSync.cs]: changed to use two maps both in memory and cluster

**Pull request merged**